streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -96,11 +98,20 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
 
             if (enableSpuriousResultFix
                 && StreamsConfig.InternalConfig.getBoolean(
-                    context().appConfigs(),
+                    context.appConfigs(),
                     ENABLE_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX,
                     true
                 )) {
                 outerJoinWindowStore = outerJoinWindowName.map(context::getStateStore);
+                sharedTimeTracker.nextTimeToEmit = context.currentSystemTimeMs();
+
+                final Object emitIntervalConfig = context.appConfigs()
+                    .get(InternalConfig.EMIT_INTERVAL_MS_KSTREAMS_OUTER_JOIN_SPURIOUS_RESULTS_FIX);
+                if (emitIntervalConfig instanceof Number) {
+                    sharedTimeTracker.setEmitInterval(((Number) emitIntervalConfig).longValue());
+                } else if (emitIntervalConfig instanceof String) {
+                    sharedTimeTracker.setEmitInterval(Long.parseLong((String) emitIntervalConfig));
+                }
```

Comment:
Shouldn't be good to move this code inside `StreamsConfig.InternalConfig`? I did that for the `getBoolean` so I could re-use it in other places. This is a good candidate for internal configs.